velox/exec/HashProbe.cpp

```diff
@@ -169,11 +169,12 @@ BlockingReason HashProbe::isBlocked(ContinueFuture* future) {
       // Build side is empty. Inner, right and semi joins return nothing in this
       // case, hence, we can terminate the pipeline early.
       if (isInnerJoin(joinType_) || isSemiJoin(joinType_) ||
-          isRightJoin(joinType_)) {
+          isRightJoin(joinType_) || isRightSemiJoin(joinType_)) {
```

Comment:
perhaps, update the comment to replace "semi joins" with "left semi joins".

Reply:
I see. This makes sense. I feel it would be easier to read the code if we changed semiJoin to leftSemiJoin. Otherwise, isSemiJoin reads as any kids of Semi join while isRightSemiJoin reads as a subset of just the Right semi joins.

Reply:
Ok. I will first add a PR to change semi to left semi. That will make this review easier.